README.md
# aot_dailogs
 A flutter package that provides commonly used dialog boxes

# usage
 [Example] (https://github.com/algoocean-technologies-pvt-ltd/aot-dialogs/blob/main/aot_dailogs/Example/example_app.dart)

 To use this package : *add the dependency to your [pubspec.yaml] file.
 
 ```yaml
    dependencies:
        flutter:
            sdk: flutter
        aot_dailogs:
```

## Add to your dart file

```dart

import 'package:aot_dailogs/aot_dailogs.dart';
import 'package:aot_dailogs/images.dart';
import 'package:flutter/material.dart';
class DialogBoxDemo extends StatefulWidget {
  @override
  State<StatefulWidget> createState() {
    return new DialogBoxDemoState();
  }
}

class DialogBoxDemoState extends State<DialogBoxDemo> {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
        appBar: new AppBar(
          title: new Text("AOT-Dialog Box"),
        ),
        body: new Center(
          child: new Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              new Text(
                'Click',
                style: new TextStyle(
                  fontWeight: FontWeight.bold,
                  fontSize: 160.0,
                  fontFamily: 'Roboto',
                ),
              ),
              new Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: <Widget>[
                  new RaisedButton(
                    padding: const EdgeInsets.all(8.0),
                    textColor: Colors.white,
                    color: Colors.blue,
                    onPressed: () {
                      AOTDialogs.okDialog(
                          context, 'Success', CustomImages.checkImage);
                    },
                    child: new Text("Confirm Dialog"),
                  ),
                  new RaisedButton(
                    onPressed: () {
                      AOTDialogs.okDialog(
                          context, 'Success', CustomImages.crossImage);
                    },
                    textColor: Colors.white,
                    color: Colors.red,
                    padding: const EdgeInsets.all(8.0),
                    child: new Text(
                      "Okay Dialog",
                    ),
                  ),
                ],
              )
            ],
          ),
        ));
  }
}

```

## Getting Started

This project is a starting point for a Dart
[package](https://flutter.dev/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.dev/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.


CHANGELOG.md
## [0.0.1] - TODO: Add release date.

* TODO: AOT Dialogs includes three commonly used dialog boxes used in flutter projects.


aot_dailogs/test/aot_dailogs_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:aot_dailogs/aot_dailogs.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
    expect(() => calculator.addOne(null), throwsNoSuchMethodError);
  });
}


lib/aot_dailogs.dart
library aot_dailogs;

import 'dart:io';
import 'package:flutter/material.dart';
import 'package:flutter/painting.dart';
import 'package:flutter/widgets.dart';

enum ConfirmAction { CANCEL, ACCEPT }
enum ExitAction { YES, NO }

class AOTDialogs {
  static Future<ConfirmAction> confirmDialog(BuildContext context) {
    return showDialog<ConfirmAction>(
      context: context,
      builder: (BuildContext context) {
        return AlertDialog(
          shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.all(Radius.circular(32.0))),
          title: Center(
            child: Column(
              children: <Widget>[
                Container(
                  child: Image(
                    image: AssetImage(
                      'assets/icons/cool-1.png',
                    ),
                    height: 85.0,
                    width: 85.0,
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.only(top: 15.0, bottom: 15.0),
                  child: Text(
                    'Are you sure ?',
                    style: TextStyle(
                        fontFamily: 'sedgwick',
                        fontWeight: FontWeight.normal,
                        fontSize: 16.0),
                    textAlign: TextAlign.center,
                  ),
                ),
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceAround,
                  children: <Widget>[
                    Container(
                      child: RaisedButton(
                        onPressed: () {
                          Navigator.of(context).pop(ConfirmAction.ACCEPT);
                        },
                        padding: EdgeInsets.only(right: 15, left: 15),
                        color: Theme.of(context).primaryColor,
                        shape: new RoundedRectangleBorder(
                          borderRadius: new BorderRadius.circular(30.0),
                        ),
                        child: Text(
                          "CONTINUE",
                          style: TextStyle(
                              fontSize: 14.0,
                              fontFamily: 'roboto',
                              color: Colors.white),
                        ),
                      ),
                    ),
                    Container(
                      child: RaisedButton(
                        onPressed: () {
                          Navigator.of(context).pop(ConfirmAction.CANCEL);
                        },
                        color: Colors.red,
                        padding: EdgeInsets.only(right: 5, left: 10),
                        shape: new RoundedRectangleBorder(
                          borderRadius: new BorderRadius.circular(30.0),
                        ),
                        child: Text(
                          "CANCEL",
                          style: TextStyle(
                              fontSize: 14.0,
                              fontFamily: 'roboto',
                              color: Colors.white),
                        ),
                      ),
                    ),
                  ],
                ),
              ],
            ),
          ),
        );
      },
    );
  }

  static Future<Widget> okDialog(
      BuildContext context, String message, AssetImage image) {
    return showDialog(
      context: context,
      builder: (BuildContext context) {
        return AlertDialog(
          shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.all(Radius.circular(32.0))),
          title: Center(
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: <Widget>[
                Container(
                  child: Image(
                    image: image,
                    height: 85.0,
                    width: 85.0,
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.only(top: 15.0, bottom: 15.0),
                  child: Text(
                    message,
                    style: TextStyle(
                        fontFamily: 'sedgwick',
                        fontWeight: FontWeight.w400,
                        fontSize: 16.0),
                    textAlign: TextAlign.center,
                  ),
                ),
                Container(
                  child: RaisedButton(
                    onPressed: () {
                      Navigator.pop(context);
                    },
                    padding: EdgeInsets.only(left: 50, right: 50),
                    color: Theme.of(context).primaryColor,
                    shape: new RoundedRectangleBorder(
                        borderRadius: new BorderRadius.circular(30.0)),
                    child: Text(
                      "OK",
                      style: TextStyle(
                          fontSize: 18.0,
                          fontFamily: 'roboto',
                          color: Colors.white),
                    ),
                  ),
                ),
              ],
            ),
          ),
        );
      },
    );
  }

  static Future<ExitAction> exitDialog(BuildContext context) {
    return showDialog<ExitAction>(
      context: context,
      builder: (BuildContext context) {
        return AlertDialog(
          shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.all(Radius.circular(32.0))),
          title: Center(
            child: Column(
              children: <Widget>[
                Container(
                  child: Image(
                    image: AssetImage(
                      'assets/images/Logo.png',
                    ),
                    height: 85.0,
                    width: 85.0,
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.only(top: 15.0, bottom: 15.0),
                  child: Text(
                    'Are you sure you want to exit?',
                    style: TextStyle(
                        fontFamily: 'sedgwick',
                        fontWeight: FontWeight.normal,
                        fontSize: 16.0),
                    textAlign: TextAlign.center,
                  ),
                ),
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceAround,
                  children: <Widget>[
                    Container(
                      child: RaisedButton(
                        onPressed: () {
                          exit(0);
                        },
                        padding: EdgeInsets.only(right: 15, left: 15),
                        color: Theme.of(context).primaryColor,
                        shape: new RoundedRectangleBorder(
                          borderRadius: new BorderRadius.circular(30.0),
                        ),
                        child: Text(
                          "YES",
                          style: TextStyle(
                              fontSize: 14.0,
                              fontFamily: 'roboto',
                              color: Colors.white),
                        ),
                      ),
                    ),
                    Container(
                      child: RaisedButton(
                        onPressed: () {
                          Navigator.of(context).pop(ExitAction.NO);
                        },
                        color: Colors.red,
                        padding: EdgeInsets.only(right: 25, left: 25),
                        shape: new RoundedRectangleBorder(
                          borderRadius: new BorderRadius.circular(30.0),
                        ),
                        child: Text(
                          "NO",
                          style: TextStyle(
                              fontSize: 14.0,
                              fontFamily: 'roboto',
                              color: Colors.white),
                        ),
                      ),
                    ),
                  ],
                ),
              ],
            ),
          ),
        );
      },
    );
  }
}


Example/example_app.dart

import 'package:aot_dailogs/aot_dailogs.dart';
import 'package:aot_dailogs/images.dart';
import 'package:flutter/material.dart';
class DialogBoxDemo extends StatefulWidget {
  @override
  State<StatefulWidget> createState() {
    return new DialogBoxDemoState();
  }
}

class DialogBoxDemoState extends State<DialogBoxDemo> {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
        appBar: new AppBar(
          title: new Text("AOT-Dialog Box"),
        ),
        body: new Center(
          child: new Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              new Text(
                'Click',
                style: new TextStyle(
                  fontWeight: FontWeight.bold,
                  fontSize: 160.0,
                  fontFamily: 'Roboto',
                ),
              ),
              new Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: <Widget>[
                  new RaisedButton(
                    padding: const EdgeInsets.all(8.0),
                    textColor: Colors.white,
                    color: Colors.blue,
                    onPressed: () {
                      AOTDialogs.okDialog(
                          context, 'Success', CustomImages.checkImage);
                    },
                    child: new Text("Confirm Dialog"),
                  ),
                  new RaisedButton(
                    onPressed: () {
                      AOTDialogs.okDialog(
                          context, 'Success', CustomImages.crossImage);
                    },
                    textColor: Colors.white,
                    color: Colors.red,
                    padding: const EdgeInsets.all(8.0),
                    child: new Text(
                      "Okay Dialog",
                    ),
                  ),
                ],
              )
            ],
          ),
        ));
  }
}